### ISSUE TITLE
BSS breaks shortened osu.ppy.sh links

### ISSUE THREAD

**Issue by iptq** [ppy/osu-web#7219]
The website shortens URLs starting with osu.ppy.sh, so it only displays the path, like this:

![image](https://user-images.githubusercontent.com/1700346/108067448-b3bef000-7058-11eb-86ce-f0f962620d91.png)

Problem is when BSS fetches metadata for the beatmap description, only the shortened version appears:

![image](https://user-images.githubusercontent.com/1700346/108067525-d3561880-7058-11eb-991d-3b760c90b680.png)

When I then press upload, the beatmap description is replaced with the shortened version, which is _not_ a URL, so the description is then replaced with:

![image](https://user-images.githubusercontent.com/1700346/108067612-f4b70480-7058-11eb-829a-0336b40ffe7c.png)

**Comment by Coppertine** [comment]
How to Reproduce:
1. Add a https://osu.ppy.sh/ link to the description on the website view of a beatmap. This should shorten the link you placed in.
2. Upload the mapset without touching anything on the description. The editor only shows the shortened link when  about to upload.
3. Check back on the website description. The link should now be rendered useless.

**Comment by peppy** [comment]
This looks to happen because editing on the website will add html content to the bbcode content, which then gets stripped on submission from BSS. @nanaya is this just how things work with forum posts from `osu-web`?

Not sure on the best way to handle things in this case. Might have to leave this one be until we get around to refactoring/rewriting BSS, since I don't particularly want to change the phpbb legacy code which is being used to do this submission process..

**Comment by nanaya** [comment]
yes as that's how the original phpbb code did it. It should match the legacy behavior. I think.

**Comment by peppy** [comment]
Hmm, weird. Wonder why it's getting stripped then. Are all URLs converted to HTML?

**Comment by nanaya** [comment]
apparently this one is due to osu-web doesn't have the exact same behaviour. The `href` attribute value probably was originally quoted using double quote but the osu-web one is single quoted and broke the matching in the bss related function (search for `bbcode_htm`). I don't know why it's using its own special parser instead of BBCodeFromDB's `toEditor()` in the first place...

**Comment by peppy** [comment]
@nanaya Can we just rewrite all these weird shortcut links in the posts table and call it a day? if so feel free to go for it.

**Comment by nanaya** [comment]
hmm, if I read it correct, it may parse the url incorrectly anyway because of the `rel="nofollow"` attribute which I don't think we want to remove.

...or maybe I can just remove it for internal url and keep it for the rest.
